### DISCUSSION TITLE
Updating Jetson devices in the field

### DISCUSSION THREAD

**Discussion by Jesseeee** [OE4T/meta-tegra#1201] [Q&A]
Hello there,

I've got a question I just can't find myself a good answer to, we're currently working on a carrier board for an AGX Xavier, which should be done soon, in the meantime we're working on the dev kit and managed to do all the kernel work write all our necessary application stuff,  currently all based on Jetpack 5.0.1, but we should be able to easily transition to 5.1 if we update all our yocto stuff, we got a pretty well working build we can flash on our dev kit (big shoutout to the tegrademo repo, provided a very solid start!) so far so good, everything we need seems to work, but as we're going to have to start shipping these the question is now popping up as to how we will update these.

It's very likely we will want to do kernel/dts/bootloader updates, generally we also use a rootfs with fuse and use A/B for both, now I started looking through the nvidia docs and they speak of the BUP and their tooling surrounding it and then that rootfs A/B should also be available now, then I looked at mender but that doesn't seem to be supported for this jetpack version.
So is there a flow described somewhere on how we could implement A/B bootloader/kernel (BUP?) and rootfs updates on the latest jetpack and what tools we'd best already pack into our initial flash to make this possible?

Ps: are there any guides on how the jetson partitioning works and where we can tweak it for e.g. putting the overlay in it's own partition?

Thanks! And also a big thanks for all the work done in this repo and the tegrademo, really helped us out a lot in developing something quickly and gave some good insights into the weird little things they do on the jetson platform.

**Comment by dwalkes** [comment]
Hi @Jesseeee 
> So is there a flow described somewhere on how we could implement A/B bootloader/kernel (BUP?) and rootfs updates on the latest jetpack

There's a thread [here](https://matrix.to/#/!mKaltPMAHUgQwPUKcH:gitter.im/$LhuRuYZnDE4ugq-FUxFDaiqan3y3XwvuxXOs5UVKMZM?via=gitter.im&via=matrix.org&via=0m.ax) about this.  NVIDIA supports a Capsule Update which should support A/B bootloader updates on Jetpack 5.1 with [this mechanism](https://docs.nvidia.com/jetson/archives/r35.2.1/DeveloperGuide/text/SD/Bootloader/UpdateAndRedundancy.html#generating-the-capsule-update-payload).  I've verified this works with stock L4T but haven't been successful in getting it to work with a meta-tegra image yet.

I've opened https://matrix.to/#/#OE4T_jetpack-5-device-update:gitter.im to track discussions about this if you'd like to join there.

**Comment by dwalkes** [comment]
>  are there any guides on how the jetson partitioning works and where we can tweak it for e.g. putting the overlay in it's own partition?

I usually recommend to look at tegrademo-mender and the flash layouts in https://github.com/mendersoftware/meta-mender-community/tree/kirkstone/meta-mender-tegra/recipes-bsp/tegra-binaries - although I don't know that we have a good example for Jetpack 5 yet.
